services/121-service/src/roles.guard.ts

```diff
@@ -44,7 +47,11 @@ export class RolesGuard implements CanActivate {
       );
       const user = await this.userService.findById(decoded.id);
 
-      hasAccess = roles.includes(user.user.role);
+      const userRoles = user.user.roles.map(role => role.role);
+      const overlappingRoles = endpointRoles.filter(role =>
+        userRoles.includes(role),
+      );
+      hasAccess = overlappingRoles.length > 0;
```

Comment:
This can be moved into the `user.service.ts`, I think.
Similar to how the HO-Portal handles it.